mne/channels/montage.py

```diff
@@ -768,18 +711,55 @@ def read_dig_montage(hsp=None, hpi=None, elp=None, point_names=None,
         # ELP
         if isinstance(elp, str):
             elp = _read_dig_points(elp, unit=unit)
-        elif elp is not None and scale[unit]:
-            elp *= scale[unit]
-        coord_frame = 'unknown'
+        elif elp is not None:
+            elp *= _scaling
+
+        data = Bunch(
+            nasion=None, lpa=None, rpa=None,
+            hsp=hsp, elp=elp, coord_frame='unknown',
+            dig_ch_pos=None,  # silently overwritten if kwarg != None
+
+            # values untouched from Kwargs
+            hpi=hpi, point_names=point_names,
+        )
 
-    # Transform digitizer coordinates to neuromag space
-    out = DigMontage(hsp, hpi, elp, point_names, fids[0], fids[1], fids[2],
-                     dig_ch_pos=dig_ch_pos, coord_frame=coord_frame)
     if fif is None and transform:  # only need to do this for non-Neuromag
-        out.transform_to_head()
+        data = _transform_to_head_call(data)
+    else:
+        pass  # noqa
+
     if dev_head_t:
-        out.compute_dev_head_t()
-    return out
+        # XXX: hpi is only defined if given by the user or fif.
+        #      So, `dev_head_t` for egi will always hit this error, 'cos we
+        #      force hip=None
+        if data.elp is None or hpi is None:
+            raise RuntimeError('must have both elp and hpi to compute the '
+                               'device to head transform')
+        else:
+            data['dev_head_t'] = fit_matched_points(
+                tgt_pts=data.elp, src_pts=data.hpi, out='trans'
+            )
+    else:
+        data['dev_head_t'] = None
+
+    # XXX: This should work but it does not!!! even if the keys are matching
+    # return DigMontage(*data)
+    __init__expected_keys = set([
+        'hsp', 'hpi', 'elp', 'point_names', 'nasion', 'lpa', 'rpa',
+        'dev_head_t', 'dig_ch_pos', 'coord_frame',
+    ])
+    assert set(data.keys()) == __init__expected_keys
```

Comment:
Any idea why `return DigMontage(*data)` would break the testsuite?

cc: @larsoner

Reply:
yep it was missing *. `return DigMontage(**data)`.